### PR TITLE
ci: Add commitlint action

### PR DESCRIPTION
**Pull request by eiri**
## PR summary

This is just a test for https://github.com/wagoid/commitlint-github-action

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by eiri** [comment]
For posterity:

tl;dr - It works, it is easy to setup, but it's kind of PITA to use.

It wants all the commits in PR to pass, not only the last one, so if you made a mistake you have to rebase and rewrite history and force push to make it happy, which is not the most user friendly approach, though I understand why it made this way.

Still I'd prefer to have my messy history during the development and fix it on a merge instead of stopping and fixing something not exactly related to a current work during the dev process. Plus it eats up to our github actions budget 😞

So to sum it up: I think following https://www.conventionalcommits.org/en/v1.0.0 is a right thing to do in general and having some checks for that would be nice to have, but they should be done locally as a pre-commit hook rather than post-commit as part of CI.
